### PR TITLE
Update Developer Guide with the latest documentation

### PR DESCRIPTION
**Pull request by iamtr**
This PR will update the Developer Guide, which consists of the latest documentation.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -237,7 +237,72 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+#### Implementation
+
+The add patient feature is facilitated by `AddCommand`, `AddCommandParser` and `Person`. 
+
+Given below is an example usage scenario and how the add patient feature behaves at each step.
+
+Step 1. The user launches the application for the first time.
+
+Step 2. The user executes an Add Command (e.g. 'add n\Alice ad\01/01/2022 dob\01/01/2002 ward\WA') to add a new patient to the address book.
```
Comment (erv-teo) [inline docs/DeveloperGuide.md]: the add command here has some errors
1. doesn't have the ic\ field
2. ward\ should just be w\

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -237,7 +237,72 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+Given below is an example usage scenario and how the add patient feature behaves at each step.
+
+Step 1. The user launches the application for the first time.
+
+Step 2. The user executes an Add Command (e.g. 'add n\Alice ad\01/01/2022 dob\01/01/2002 ward\WA') to add a new patient to the address book. 
+
+n\: Indicates the name of the patient
+ad\: Indicates the admission date of the patient
+dob\: Indicates the date of birth of the patient
+ward\: Indicates the ward of the patient is currently in
```
Comment (erv-teo) [inline docs/DeveloperGuide.md]: Same here as above, ward\ should just  be w\ and we should also include ic\

**Review by ryanlimdx — COMMENTED** [review]
LGTM

**Review by swtan346 — APPROVED** [review]
LGTM

**Pull request merged**